Sources/Evangelium/Helpers/ReadingFactory.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  ReadingFactory.swift
+//  Evangelium
+//
+//  Created by Manuel Sánchez on 2/4/20.
+//
+
+import Foundation
+
+struct ReadingFactory {
+    static func create(from todayReadings: [Reading], with date: String) -> SwiftyReading{
+        let readings = todayReadings.filter { reading in reading.type == "reading" }
+        let psalm = todayReadings.filter { reading in reading.type == "psalm" }
+        let gospel = todayReadings.filter { reading in reading.type == "gospel" }
+        
+        if readings.count == 2 {
+            return SwiftyReading(
+                firstReading: readings.first,
+                psalm: !psalm.isEmpty ? psalm.first : nil,
```

Comment:
If you are using "filter" to have an array of (psalms, gospel or readings), when "first" is used if the array is ​​empty, the value of this property is null. It is not necessary to check if it is empty.